sematic/ui/src/components/Exception.tsx

```diff
@@ -1,21 +1,38 @@
 import { Alert, AlertTitle } from "@mui/material";
 import { ExceptionMetadata } from "../Models";
 
-export default function Exception(props: { exception: ExceptionMetadata }) {
-  const { exception } = props;
+export function Exception(props: { exception_metadata: ExceptionMetadata }) {
+  const { exception_metadata } = props;
   return (
     <Alert severity="error" icon={false}>
       <AlertTitle>
-        {exception.repr.split("\n")[exception.repr.split("\n").length - 2]}
+        {exception_metadata.repr.split("\n")[exception_metadata.repr.split("\n").length - 2]}
       </AlertTitle>
       <pre
         style={{
           whiteSpace: "pre-wrap",
           overflowWrap: "anywhere",
         }}
       >
-        {exception.repr}
+        {exception_metadata.repr}
       </pre>
     </Alert>
   );
 }
+
+export function ExternalException(props: { exception_metadata: ExceptionMetadata }) {
```

Comment:
I say we use the `Exception` component above. No reason to style it differently.

Reply:
I believe the different style helps the user process the visual information when showing both the K8 failures and the code execution stack trace together:
<img width="1230" alt="image" src="https://user-images.githubusercontent.com/1894533/198363531-06d1cafc-008f-468e-b481-c08a849f263f.png">

The exact same code can't be reused because the Exception component parses out the Exception message from the stack trace. In the case of K8 errors, there is not stack trace, only plain text.

Reply:
just a note on the actual example shown here: in the case of a normal calculator error, we should not show a k8s error display as well. I think displaying k8s errors should be reserved for the case where k8s was the *cause* of the run being failed. If there was a calculator error that we detected, we can safely assume that was the cause.

Reply:
⬆️ This. Is there a case when we will have both errors, and both should be displayed?
If no, then we can use the same styling.

Additionally, when it is a user code error, the user likely recognizes the error and knows what to do. When it it a Kubernetes error, they are probably a bit more lost. We should add an additional message explaining the potential reason for it and directing people to our Discord for help.

Reply:
The content of that specific example is not actually representative; it was obtained during development. I used it to show how the rendering would look like because I had it on hand and because it's tedious to get all the manual testing scenarios just right in these error cases.

This code under review does not show the generic pod error message whenever a code execution error kills the pod. Apologies for the misdirection.

Relevant example of only an external error:
<img width="1134" alt="image" src="https://user-images.githubusercontent.com/1894533/198753847-09bc970c-561e-4143-87b6-29b855a0cfe8.png">

Relevant example of both a code execution error and an external error:
![image](https://user-images.githubusercontent.com/1894533/198753796-8171d551-11e2-438d-ae66-dc61589d4141.png)

This case when both occur could theoretically happen when an external event affects the container causing the code execution to fail and also cause the container to fail.

Also, the current behavior is that if a code execution error determines a retry, its original stack trace will be carried over to the retry run in the UI. Given this, I believe we should not inhibit showing a later pod failure error due to a previous code execution error.
I believe this behavior should be improved more sustainably as part of #239 (instead of just erasing the information about a previous try's error).

Reply:
Added a Discord help message. I chose to render it only in the UI, as generating this in the backend and saving this message in the DB seemed incorrect behavior.

I implemented it in an atrocious way; I welcome suggestions of how to improve this.

![image](https://user-images.githubusercontent.com/1894533/198754598-87ec8dc4-9721-4030-a183-cfbf969e5cb8.png)

Reply:
Thought about it some more, and I think we should fix this at a different layer. Filed #276